modules/devices.c

```diff
@@ -195,12 +195,142 @@ gchar *get_memory_total(void)
+  { "Beta",      "Q1 2012",    "B (Beta)",            "?",      "256MB",    "(Beta board)" },
+  { "0002",      "Q1 2012",    "B",                   "1.0",    "256MB",    "?" },
+  { "0003",      "Q3 2012",    "B (ECN0001)",         "1.0",    "256MB",    "(Fuses mod and D14 removed)" },
+  { "0004",      "Q3 2012",    "B",                   "2.0",    "256MB",    "Sony"    },
+  { "0005",      "Q4 2012",    "B",                   "2.0",    "256MB",    "Qisda"   },
+  { "0006",      "Q4 2012",    "B",                   "2.0",    "256MB",    "Egoman"  },
+  { "0007",      "Q1 2013",    "A",                   "2.0",    "256MB",    "Egoman"  },
+  { "0008",      "Q1 2013",    "A",                   "2.0",    "256MB",    "Sony"    },
+  { "0009",      "Q1 2013",    "A",                   "2.0",    "256MB",    "Qisda"   },
+  { "000d",      "Q4 2012",    "B",                   "2.0",    "512MB",    "Egoman" },
+  { "000e",      "Q4 2012",    "B",                   "2.0",    "512MB",    "Sony" },
+  { "000f",      "Q4 2012",    "B",                   "2.0",    "512MB",    "Qisda" },
+  { "0010",      "Q3 2014",    "B+",                  "1.0",    "512MB",    "Sony" },
+  { "0011",      "Q2 2014",    "Compute Module 1",    "1.0",    "512MB",    "Sony" },
+  { "0012",      "Q4 2014",    "A+",                  "1.1",    "256MB",    "Sony" },
+  { "0013",      "Q1 2015",    "B+",                  "1.2",    "512MB",    "?" },
+  { "0014",      "Q2 2014",    "Compute Module 1",    "1.0",    "512MB",    "Embest" },
+  { "0015",      "?",          "A+",                  "1.1",    "256MB/512MB",    "Embest" },
+  { "a01040",    "Unknown",    "2 Model B",           "1.0",    "1GB",      "Sony" },
+  { "a01041",    "Q1 2015",    "2 Model B",           "1.1",    "1GB",      "Sony" },
+  { "a21041",    "Q1 2015",    "2 Model B",           "1.1",    "1GB",      "Embest" },
+  { "a22042",    "Q3 2016",    "2 Model B",           "1.2",    "1GB",      "Embest" },  /* (with BCM2837) */
+  { "900021",    "Q3 2016",    "A+",                  "1.1",    "512MB",    "Sony" },
+  { "900032",    "Q2 2016?",    "B+",                 "1.2",    "512MB",    "Sony" },
+  { "900092",    "Q4 2015",    "Zero",                "1.2",    "512MB",    "Sony" },
+  { "900093",    "Q2 2016",    "Zero",                "1.3",    "512MB",    "Sony" },
+  { "920093",    "Q4 2016?",   "Zero",                "1.3",    "512MB",    "Embest" },
+  { "9000c1",    "Q1 2017",    "Zero W",              "1.1",    "512MB",    "Sony" },
+  { "a02082",    "Q1 2016",    "3 Model B",           "1.2",    "1GB",      "Sony" },
+  { "a020a0",    "Q1 2017",    "Compute Module 3 or CM3 Lite",    "1.0",    "1GB",    "Sony" },
+  { "a22082",    "Q1 2016",    "3 Model B",           "1.2",    "1GB",    "Embest" },
+  { "a32082",    "Q4 2016",    "3 Model B",           "1.2",    "1GB",    "Sony Japan" },
+  { NULL, NULL, NULL, NULL, NULL, NULL }
+};
+
+gchar *dt_get_model() {
```

Comment:
You can use `g_file_get_contents()` instead of everything inside this function.  Please also make this a `static` function so it's not exported.